app/persisters/indexing_persister.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+class IndexingPersister
+  attr_reader :persister, :index_persister, :workflow_decorator
+  delegate :adapter, to: :composite_persister
+  def initialize(persister:, index_persister:, workflow_decorator: SimpleDelegator)
+    @persister = persister
+    @index_persister = index_persister
+    @workflow_decorator = workflow_decorator
+  end
+
+  def save(model:)
+    composite_persister.save(model: model)
+  end
+
+  def save_all(models:)
+    composite_persister.save_all(models: models)
+  end
+
+  def delete(model:)
+    composite_persister.delete(model: model)
+  end
+
+  def buffer_into_index
+    buffered_persister.with_buffer do |persist, buffer|
+      persist = workflow_decorator.new(persist)
+      yield persist
+      buffer.persister.deletes.uniq(&:id).each do |delete|
+        index_persister.delete(model: delete)
+      end
+      index_persister.save_all(models: buffer.query_service.find_all)
+    end
+  end
```

Comment:
This is a mildly exotic pattern. Maybe the buffered persister sheds more light, but I'm kind of confused by this method.

We wrap the persister with any workflow observers/chain of command and immediately yield it (so the caller can do what they need in a block), then purge anything deleted from the index, then reindex <all of something>.

I think that's pretty close to _what_ is happening, but maybe a bit of explanation of why this works this way (what the buffer actually is), and why this is an `IndexingPersister` responsibility would be good? (I'm sort of seeing a bit of a BufferedPersister template method here, but maybe not...)